src/covariance.h

```diff
@@ -157,15 +170,40 @@ matrix<T> get_covariance_lower_chol(const vector<T>& theta, int n_visits, std::s
   return result;
 }
 
+// Creates a new spatial covariance cholesky.
+template <class T>
+matrix<T> get_spatial_covariance_lower_chol(const vector<T>& theta, const matrix<T>& distance, std::string cov_type) {
+  matrix<T> result;
+  if (cov_type == "gp_exp") {
+    result = get_spatial_exponential<T>(theta, distance);
+  } else {
+    Rf_error(("Unknown covariance type '" + cov_type + "'.").c_str());
+  }
+  return result;
+}
+
 // Creates a grouped correlation object dynamically.
 template <class T>
-matrix<T> get_cov_lower_chol_grouped(const vector<T>& theta, int n_visits, std::string cov_type, int n_groups) {
+matrix<T> get_cov_lower_chol_grouped(const vector<T>& theta, int n_visits, std::string cov_type, int n_groups, int spatial) {
+  if (spatial == 1) {
+    n_visits = 2;
+  }
   matrix<T> result(n_visits * n_groups, n_visits);
   int covariance_size = theta.size() / n_groups;
-  for (int i = 0; i < n_groups; i++) {
-    matrix<T> lower_chol = get_covariance_lower_chol<T>(theta.segment(i * covariance_size, covariance_size), n_visits, cov_type);
-    result << result.block(0, 0, n_visits * i, n_visits), lower_chol;
+  if (spatial == 1) {
+    matrix<T> standard_dist(2, 2);
+    standard_dist << 0, 1, 1, 0;
```

Comment:
why do you take the 2x2 identity matrix here?

Reply:
following SAS outputs, we have subject-level covariance matrix. I think it is enough to display the covariance matrix for the standard distance matrix c(0,1,1,0), it is not identity matrix.